rllib/env/base_env.py

```diff
@@ -161,7 +161,7 @@ def send_actions(self, action_dict: MultiEnvDict) -> None:
 
     @PublicAPI
     def try_reset(self,
-                  env_id: Optional[EnvID] = None) -> Optional[MultiAgentDict]:
+                  env_id: Optional[EnvID] = None) -> Optional[MultiEnvDict]:
```

Comment:
seems ok, but this is @PublicAPI, so we would again go through a deprecation process etc ...
I guess just do it in your validation code?
or alternatively, we can add as_dict=False, and return Optional[Union[MultiEnvDict, MultiAgentDict]]. but that feels kinda ugly ...
up to you man.

Reply:
Wait, this should be `Union[MultiAgentDict, MultiEnvDict]`, depending on whether env_id is provided (MultiAgentDict) or not (MultiEnvDict)?

Ok, but I'm also seeing that in `_MultiAgentEnvToBaseEnv`, we don't handle the `env_id=None` case properly:

```
    @override(BaseEnv)
    def try_reset(self,
                  env_id: Optional[EnvID] = None) -> Optional[MultiAgentDict]:
        obs = self.env_states[env_id].reset()
```

^ KeyError if env_id=None.

Reply:
@gjoliver I actually didn't change the return types for the original wrappers, so I think that what I will do instead is add the return type as Optional[Union[MultiEnvDict, MultiAgentDict]]

BUT

only because all of the old env wrappers return a `MultiAgentDict` and all the new ones will return a `MultiEnvDict` and not because env_id is provided or not. 

@sven1977 is that functionality necessary, that we reset all environments when called with `None`? It isn't already supported, although I can easily support it, However, it seems like a footgun